internal/connector/connector.go

```diff
@@ -233,6 +238,7 @@ func Run(ctx context.Context, options Options) error {
 	}
 
 	go func() {
+		// TODO: failing to bind should cause process to exit
```

Comment:
This TODO will be addressed along with the other one I commented on. It's all related to the goroutine lifetime.